{
    "title": "UMF Error Json",
    "fieldStartBits": {
        "ReservedBits": 22,
        "Classes": 14,
        "IDs": 0
    },
    "classes": {
        "None": {
            "id": 0,
            "alias": "",
            "description": "No error and PI errors predating UMF"
        },
        "Debug": {
            "id": 255,
            "alias": "DBG",
            "description": "Debug messages for internal PI usage."
        },
        "Info": {
            "id": 1,
            "alias": "INFO",
            "description": "Information. No further action required."
        },
        "Warning": {
            "id": 2,
            "alias": "WARN",
            "description": "Warning. Action/inspection recommended."
        },
        "Command": {
            "id": 3,
            "alias": "CMD",
            "description": "Command input error. Caused due to improper GCS-commands. Only triggered on the current interface"
        },
        "Critical": {
            "id": 4,
            "alias": "CRIT",
            "description": "Runtime error that wasn't triggered by an improper command input or by a motion error in the Drive State Machine. Action required."
        },
        "Motion": {
            "id": 5,
            "alias": "MOT",
            "description": "Motion error of axis. Triggers the error handling in the Drive State Machine (Transition: operation enabled -> fault reaction -> fault)"
        }
    },
    "modules": {
        "Zero": {
            "id": 0,
            "alias": "",
            "description": "Reserved for old Error Sets and No Error"
        },
        "One": {
            "id": 1,
            "alias": "",
            "description": "Reserved for old Error Sets"
        },
        "Command": {
            "id": 2,
            "alias": "CMD",
            "description": "Command Module"
        },
        "Macro": {
            "id": 3,
            "alias": "MACRO",
            "description": "Macro Module"
        },
        "Parameter": {
            "id": 4,
            "alias": "PARAM",
            "description": "Parameter Module"
        },
        "Motion": {
            "id": 5,
            "alias": "MOT",
            "description": "Motion Module"
        },
        "DataRecorder": {
            "id": 6,
            "alias": "REC",
            "description": "Data Recorder Module"
        },
        "Sensor": {
            "id": 7,
            "alias": "SENS",
            "description": "Sensor Module"
        },
        "Com": {
            "id": 8,
            "alias": "COM",
            "description": "Com Module"
        },
        "System": {
            "id": 9,
            "alias": "SYS",
            "description": "System Module"
        }
    },
    "errors": {
        "PI_ERROR_NO_ERROR": {
            "id": 0,
            "status": "released",
            "module": "Zero",
            "class": ["None"],
            "description": "No error",
            "internal_usage": "",
            "cmds": [],
            "obsolete_id": 0,
            "pageId": 0,
            "obsolete_error_codes": [0]
        },
        "PI_ERROR_$MODULE_$CLASS_NUMBER_OF_ARGUMENTS": {
            "id": 2,
            "status": "released",
            "module": "Command",
            "class": ["Command"],
            "description": "Wrong number of arguments",
            "internal_usage": "",
            "cmds": ["*IDNq","CPA","CSVq","EAX","EAXq","ERRq","FRF","FRFq","FWC","FWCq","FWE","FWEq","FWRq","HLPq","HLT","LOGq","MOV","MOVq","OCV","OCVq","ONTq","POSq","REC_RATE","REC_START","REC_STOP","REC_TRACE","REC_TRG","RECq_DAT","RECq_NUM","RECq_RATE","RECq_STATE","RECq_TRACE","RECq_TRG","RES","SAIq","SAM","SAMq","SMRq","SMV","SMVq","SPC","SPV","SPVq","STP","STVq","SVO","SVOq","UCL","UCLq","USGq","USGq_CMD","USGq_HW","USGq_PAM","USGq_PROP","USGq_SYS","VERq","RBT", "REC"],
            "obsolete_id": 1,
            "obsolete_define": "PI_ERROR_CMD_NUMBER_OF_ARGUMENTS",
            "pageId": 225674723,
            "obsolete_error_codes": [8961]
        },
        "PI_ERROR_$MODULE_$CLASS_UNKNOWN_COMMAND": {
            "id": 3,
            "status": "released",
            "module": "Command",
            "class": ["Command"],
            "description": "Unknown command",
            "internal_usage": "",
            "cmds": [],
            "obsolete_id": 2,
            "obsolete_define": "PI_ERROR_CMD_UNKNOWN_COMMAND",
            "pageId": 225674724,
            "obsolete_error_codes": [8962]
        },
        "PI_ERROR_$MODULE_$CLASS_COMMAND_LEVEL_TOO_LOW_FOR_COMMAND_ACCESS": {
            "id": 4,
            "status": "released",
            "module": "Command",
            "class": ["Command"],
            "description": "User command level too low for command access",
            "internal_usage": "",
            "cmds": ["SPC","USGq_CMD","FWC","FWCq","FWE","FWEq","FWRq"],
            "obsolete_id": 5,
            "obsolete_define": "PI_ERROR_CMD_COMMAND_LEVEL_TOO_LOW_FOR_COMMAND_ACCESS",
            "pageId": 225674721,
            "obsolete_error_codes": [8965]
        },
        "PI_ERROR_$MODULE_$CLASS_INVALID_PWD": {
            "id": 5,
            "status": "released",
            "module": "Command",
            "class": ["Command"],
            "description": "Invalid password",
            "internal_usage": "",
            "cmds": ["UCL"],
            "obsolete_id": 10,
            "obsolete_define": "PI_ERROR_CMD_INVALID_PWD",
            "pageId": 225674722,
            "obsolete_error_codes": [8970]
        },
        "PI_ERROR_$MODULE_$CLASS_UNKNOWN_SECTION_COMMAND": {
            "id": 6,
            "status": "released",
            "module": "Command",
            "class": ["Command"],
            "description": "Unknown section command",
            "internal_usage": "",
            "cmds": ["REC","RECq","USGq"],
            "obsolete_id": 12,
            "obsolete_define": "PI_ERROR_CMD_UNKNOWN_SECTION_COMMAND",
            "pageId": 240060887,
            "obsolete_error_codes": [8972]
        },
        "PI_ERROR_$MODULE_$CLASS_INVALID_CHAR": {
            "id": 7,
            "status": "released",
            "module": "Command",
            "class": ["Command"],
            "description": "Received invalid ASCII character",
            "internal_usage": "",
            "cmds": [],
            "obsolete_id": 40,
            "obsolete_define": "PI_ERROR_CMD_INVALID_CHAR",
            "pageId": 250347601,
            "obsolete_error_codes": [9000]
        },
        "PI_ERROR_$MODULE_$CLASS_STP": {
            "id": 8,
            "status": "released",
            "module": "Command",
            "class": ["Motion"],
            "description": "Stop command",
            "internal_usage": "Error triggered when calling STP, note that it doesn't appear in ERR?, but only in the LOG",
            "cmds": ["STP"],
            "obsolete_id": 110,
            "obsolete_define": "PI_ERROR_CMD_STOP",
            "pageId": 232397433,
            "obsolete_error_codes": [9582]
        },
        "PI_ERROR_$MODULE_$CLASS_WRONG_DATA_TYPE": {
            "id": 9,
            "status": "released",
            "module": "Parameter",
            "class": ["Debug", "Command"],
            "description": "Wrong data type",
            "internal_usage": "",
            "cmds": ["REC_TRACE"],
            "obsolete_id": 1,
            "obsolete_define": "PI_ERROR_PARAM_WRONG_DATA_TYPE",
            "pageId": 225674741,
            "obsolete_error_codes": [16385]
        },
        "PI_ERROR_$MODULE_$CLASS_UNKNOWN_PARAMETER_ID": {
            "id": 10,
            "status": "released",
            "module": "Parameter",
            "class": ["Command"],
            "description": "Unknown parameter ID",
            "internal_usage": "",
            "cmds": ["CPA","REC_TRACE","SPC","SPV","SPVq","USGq_PAM"],
            "obsolete_id": 2,
            "obsolete_define": "PI_ERROR_PARAM_UNKNOWN_PARAMETER_ID",
            "pageId": 225674739,
            "obsolete_error_codes": [17154]
        },
        "PI_ERROR_$MODULE_$CLASS_COMMAND_LEVEL_TOO_LOW_FOR_PARAMETER_ACCESS": {
            "id": 11,
            "status": "released",
            "module": "Parameter",
            "class": ["Command"],
            "description": "User command level too low for parameter access",
            "internal_usage": "",
            "cmds": ["CPA","REC_TRACE","SPC","SPV","SPVq","USGq_PAM"],
            "obsolete_id": 5,
            "obsolete_define": "PI_ERROR_PARAM_COMMAND_LEVEL_TOO_LOW_FOR_PARAMETER_ACCESS",
            "pageId": 225674737,
            "obsolete_error_codes": [17157]
        },
        "PI_ERROR_$MODULE_$CLASS_INVALID_VALUE": {
            "id": 12,
            "status": "released",
            "module": "Parameter",
            "class": ["Command"],
            "description": "Argument value not valid",
            "internal_usage": "",
            "cmds": ["CPA","EAX","MOV","OCV","RECq","REC_RATE","REC_TRACE","RECq_DAT","RECq_TRG","SPV","SPVq","SVO","UCL","USGq","USGq_CMD","FWE","FWRq","LOGq","SAIq","SMV","SPC"],
            "obsolete_id": 10,
            "obsolete_define": "PI_ERROR_PARAM_INVALID_VALUE",
            "pageId": 225674738,
            "obsolete_error_codes": [17162]
        },
        "PI_ERROR_$MODULE_$CLASS_WRONG_PARAMETER_TYPE": {
            "id": 13,
            "status": "released",
            "module": "Parameter",
            "class": ["Command"],
            "description": "Wrong parameter type",
            "internal_usage": "",
            "cmds": ["CPA","REC_TRACE","SPV","SPVq"],
            "obsolete_id": 11,
            "obsolete_define": "PI_ERROR_PARAM_WRONG_PARAMETER_TYPE",
            "pageId": 225674742,
            "obsolete_error_codes": [17163]
        },
        "PI_ERROR_$MODULE_$CLASS_VALUE_OUT_OF_RANGE": {
            "id": 14,
            "status": "released",
            "module": "Parameter",
            "class": ["Command"],
            "description": "Argument value out of range",
            "internal_usage": "",
            "cmds": ["CPA","FWC","FWE","LOGq","REC_RATE","REC_TRACE","RECq_DAT","SPV","SPVq","MOV","OCV","SMV","SPC"],
            "obsolete_id": 40,
            "obsolete_define": "PI_ERROR_PARAM_VALUE_OUT_OF_RANGE",
            "pageId": 225674740,
            "obsolete_error_codes": [17192]
        },
        "PI_ERROR_$MODULE_$CLASS_UNKNOWN_AXIS_ID": {
            "id": 15,
            "status": "released",
            "module": "Motion",
            "class": ["Command"],
            "description": "Unknown Axis ID",
            "internal_usage": "",
            "cmds": ["EAX","EAXq","FRF","FRFq","HLT","MOV","MOVq","OCV","OCVq","ONTq","POSq","RES","SAM","SAMq","SMRq","SMV","SMVq","SVO","SVOq"],
            "obsolete_id": 2,
            "obsolete_define": "PI_ERROR_MOTION_UNKNOWN_AXIS_ID",
            "pageId": 240060888,
            "obsolete_error_codes": [21250]
        },
        "PI_ERROR_$MODULE_$CLASS_ON_LIMIT_SWITCH": {
            "id": 16,
            "status": "released",
            "module": "Motion",
            "class": ["Command"],
            "description": "Motion not possible due to active limit switch",
            "internal_usage": "Note: in FRF it is only occurs as an asynchronous error",
            "cmds": ["FRF","MOV","OCV","SMV"],
            "obsolete_id": 7,
            "obsolete_define": "PI_ERROR_MOTION_ON_LIMIT_SWITCH",
            "pageId": 225674733,
            "obsolete_error_codes": [21255]
        },
        "PI_ERROR_$MODULE_$CLASS_INVALID_MODE_OF_OPERATION": {
            "id": 17,
            "status": "released",
            "module": "Motion",
            "class": ["Command"],
            "description": "Invalid mode of operation",
            "internal_usage": "SMR? und SMV? erzeugen den Fehler, wenn gezielt nach einer Achse gefragt wird, die keinen Schrittbetrieb unterstützt. Wenn man nach allen Achsen fragt, enthält die Antwort nur die Achsen, die Schrittbetrieb unterstützen, und es kommt kein Fehler. Begründung: Damit soll die Implementierung einfach gehalten und das Verhalten so definiert werden wie bei SMV (wo es auch nur diesen Fehler gibt, egal ob die Achse den Modus gar nicht unterstützt oder gerade bloß nicht in diesem Modus ist).",
            "cmds": ["MOV","FRF","OCV","SAM","SMRq","SMV","SMVq","SVO"],
            "obsolete_id": 10,
            "obsolete_define": "PI_ERROR_MOTION_INVALID_MODE_OF_OPERATION",
            "pageId": 225674730,
            "obsolete_error_codes": [21258]
        },
        "PI_ERROR_$MODULE_$CLASS_AXIS_NOT_REF": {
            "id": 18,
            "status": "released",
            "module": "Motion",
            "class": ["Command", "Motion"],
            "description": "Axis not referenced",
            "internal_usage": "",
            "cmds": ["SAM","SVO"],
            "obsolete_id": 11,
            "obsolete_define": "PI_ERROR_MOTION_MOVE_WITHOUT_REF",
            "pageId": 225674732,
            "obsolete_error_codes": [21259]
        },
        "PI_ERROR_$MODULE_$CLASS_INVALID_AXIS_STATE": {
            "id": 19,
            "status": "released",
            "module": "Motion",
            "class": ["Command"],
            "description": "Invalid state of axis state machine",
            "internal_usage": "",
            "cmds": ["RES"],
            "obsolete_id": 12,
            "obsolete_define": "PI_ERROR_MOTION_INVALID_AXIS_STATE",
            "pageId": 240060889,
            "obsolete_error_codes": [21260]
        },
        "PI_ERROR_$MODULE_$CLASS_TARGET_OUT_OF_RANGE": {
            "id": 20,
            "status": "released",
            "module": "Motion",
            "class": ["Command"],
            "description": "Target out of range",
            "internal_usage": "",
            "cmds": ["MOV"],
            "obsolete_id": 40,
            "obsolete_define": "PI_ERROR_MOTION_TARGET_OUT_OF_RANGE",
            "pageId": 225674736,
            "obsolete_error_codes": [21288]
        },
        "PI_ERROR_$MODULE_$CLASS_AXIS_DISABLED": {
            "id": 21,
            "status": "released",
            "module": "Motion",
            "class": ["Command"],
            "description": "Axis is disabled",
            "internal_usage": "",
            "cmds": ["FRF","MOV","OCV","SAM","SMV","SVO"],
            "obsolete_id": 200,
            "obsolete_define": "PI_ERROR_MOTION_AXIS_DISABLED",
            "pageId": 225674729,
            "obsolete_error_codes": [21448]
        },
        "PI_ERROR_$MODULE_$CLASS_FAULT_REACTION_ACTIVE": {
            "id": 22,
            "status": "released",
            "module": "Motion",
            "class": ["Command"],
            "description": "Fault reaction active on axis",
            "internal_usage": "",
            "cmds": ["MOV"],
            "obsolete_id": 201,
            "obsolete_define": "PI_ERROR_MOTION_FAULT_REACTION_ACTIVE",
            "pageId": 240060890,
            "obsolete_error_codes": [21449]
        },
        "PI_ERROR_$MODULE_$CLASS_LIMIT_SWITCH_ACTIVATED": {
            "id": 23,
            "status": "released",
            "module": "Motion",
            "class": ["Motion"],
            "description": "Axis has activated a limit switch",
            "internal_usage": "",
            "cmds": [],
            "obsolete_id": 7,
            "obsolete_define": "PI_ERROR_MOTION_LIMIT_SWITCH_ACTIVATED",
            "pageId": 225674731,
            "obsolete_error_codes": [21767]
        },
        "PI_ERROR_$MODULE_$CLASS_OVER_CURRENT_PROTECTION": {
            "id": 24,
            "status": "released",
            "module": "Motion",
            "class": ["Motion"],
            "description": "Motor current is too high, servo is switched off automatically",
            "internal_usage": "",
            "cmds": [],
            "obsolete_id": 40,
            "obsolete_define": "PI_ERROR_MOTION_OVER_CURRENT_PROTECTION",
            "pageId": 225674734,
            "obsolete_error_codes": [21800]
        },
        "PI_ERROR_$MODULE_$CLASS_OUTPUT_LIMIT": {
            "id": 25,
            "status": "released",
            "module": "Motion",
            "class": ["Warning"],
            "description": "Motor output is limited",
            "internal_usage": "",
            "cmds": [],
            "obsolete_id": 40,
            "obsolete_define": "PI_ERROR_MOTION_OUTPUT_LIMIT",
            "pageId": 0,
            "obsolete_error_codes": []
        },
        "PI_ERROR_$MODULE_$CLASS_POSITION_ERROR_TOO_LARGE": {
            "id": 26,
            "status": "released",
            "module": "Motion",
            "class": ["Motion"],
            "description": "Position error too large, servo is switched off automatically",
            "internal_usage": "",
            "cmds": [],
            "obsolete_id": 41,
            "obsolete_define": "PI_ERROR_MOTION_POSITION_ERROR_TOO_LARGE",
            "pageId": 225674735,
            "obsolete_error_codes": [21801]
        },
        "PI_ERROR_$MODULE_$CLASS_STOP": {
            "id": 27,
            "status": "released",
            "module": "Motion",
            "class": ["Motion"],
            "description": "Stop triggered by command or by error on another axis",
            "internal_usage": "Der Fehler wird für jede gestoppte Achse geworfen, wenn der Stopp durch den Befehl STP oder durch einen Motion Error auf einer anderen Achse ausgelöst wurde.",
            "cmds": [],
            "obsolete_id": 110,
            "obsolete_define": "PI_ERROR_MOTION_STOP",
            "pageId": 232397434,
            "obsolete_error_codes": [21870]
        },
        "PI_ERROR_$MODULE_$CLASS_MAX_DATA_RECORDER_NUMBER_REACHED": {
            "id": 28,
            "status": "released",
            "module": "DataRecorder",
            "class": ["Debug"],
            "description": "The max number of storable data recorders has been reached",
            "internal_usage": "",
            "cmds": [],
            "obsolete_id": 7,
            "obsolete_define": "PI_ERROR_RECORDER_MAX_DATA_RECORDER_NUMBER_REACHED",
            "pageId": 225674745,
            "obsolete_error_codes": [24583]
        },
        "PI_ERROR_$MODULE_$CLASS_ALREADY_REGISTERED": {
            "id": 29,
            "status": "released",
            "module": "DataRecorder",
            "class": ["Debug"],
            "description": "The given data recorder has already been registered",
            "internal_usage": "",
            "cmds": [],
            "obsolete_id": 20,
            "obsolete_define": "PI_ERROR_RECORDER_ALREADY_REGISTERED",
            "pageId": 225674743,
            "obsolete_error_codes": [24596]
        },
        "PI_ERROR_$MODULE_$CLASS_WRONG_FORMAT": {
            "id": 30,
            "status": "released",
            "module": "DataRecorder",
            "class": ["Command"],
            "description": "Wrong recorder format",
            "internal_usage": "",
            "cmds": ["RECq_DAT"],
            "obsolete_id": 1,
            "obsolete_define": "PI_ERROR_RECORDER_WRONG_FORMAT",
            "pageId": 240060891,
            "obsolete_error_codes": [25345]
        },
        "PI_ERROR_$MODULE_$CLASS_UNKNOWN_RECORDER_ID": {
            "id": 31,
            "status": "released",
            "module": "DataRecorder",
            "class": ["Command"],
            "description": "Unknown recorder ID",
            "internal_usage": "",
            "cmds": ["REC_RATE","REC_START","REC_STOP","REC_TRACE","REC_TRG","RECq_DAT","RECq_NUM","RECq_RATE","RECq_STATE","RECq_TRACE","RECq_TRG"],
            "obsolete_id": 2,
            "obsolete_define": "PI_ERROR_RECORDER_UNKNOWN_RECORDER_ID",
            "pageId": 240060892,
            "obsolete_error_codes": [25346]
        },
        "PI_ERROR_$MODULE_$CLASS_NOT_IN_CONFIG_MODE": {
            "id": 32,
            "status": "released",
            "module": "DataRecorder",
            "class": ["Command"],
            "description": "Data recorder is not in configuration mode",
            "internal_usage": "",
            "cmds": ["REC_RATE","REC_TRACE","REC_TRG"],
            "obsolete_id": 10,
            "obsolete_define": "PI_ERROR_RECORDER_NOT_IN_CONFIG_MODE",
            "pageId": 225674747,
            "obsolete_error_codes": [25354]
        },
        "PI_ERROR_$MODULE_$CLASS_WRONG_RECORDER_TRIGGER": {
            "id": 33,
            "status": "released",
            "module": "DataRecorder",
            "class": ["Command"],
            "description": "Wrong recorder trigger",
            "internal_usage": "",
            "cmds": ["REC_TRG"],
            "obsolete_id": 12,
            "obsolete_define": "PI_ERROR_RECORDER_WRONG_TRIGGER_ID",
            "pageId": 240060893,
            "obsolete_error_codes": [25356]
        },
        "PI_ERROR_$MODULE_$CLASS_WRONG_STARTPOINT": {
            "id": 34,
            "status": "released",
            "module": "DataRecorder",
            "class": ["Command"],
            "description": "Wrong start point",
            "internal_usage": "",
            "cmds": ["RECq_DAT"],
            "obsolete_id": 14,
            "obsolete_define": "PI_ERROR_RECORDER_WRONG_STARTPOINT",
            "pageId": 240060894,
            "obsolete_error_codes": [25358]
        },
        "PI_ERROR_$MODULE_$CLASS_WRONG_NUMPOINT": {
            "id": 35,
            "status": "released",
            "module": "DataRecorder",
            "class": ["Command"],
            "description": "Wrong number of points",
            "internal_usage": "",
            "cmds": ["RECq_DAT"],
            "obsolete_id": 15,
            "obsolete_define": "PI_ERROR_RECORDER_WRONG_NUMPOINT",
            "pageId": 240060895,
            "obsolete_error_codes": [25359]
        },
        "PI_ERROR_$MODULE_$CLASS_ALREADY_RUNNING": {
            "id": 36,
            "status": "released",
            "module": "DataRecorder",
            "class": ["Command"],
            "description": "Data recorder is already running",
            "internal_usage": "",
            "cmds": ["REC_START"],
            "obsolete_id": 20,
            "obsolete_define": "PI_ERROR_RECORDER_ALREADY_RUNNING",
            "pageId": 225674744,
            "obsolete_error_codes": [25364]
        },
        "PI_ERROR_$MODULE_$CLASS_TRACE_DOES_NOT_EXIST": {
            "id": 37,
            "status": "released",
            "module": "DataRecorder",
            "class": ["Command"],
            "description": "Specified data recorder trace does not exist",
            "internal_usage": "",
            "cmds": ["RECq_DAT","REC_TRACE","RECq_TRACE"],
            "obsolete_id": 40,
            "obsolete_define": "PI_ERROR_RECORDER_TRACE_DOES_NOT_EXIST",
            "pageId": 225674748,
            "obsolete_error_codes": [25384]
        },
        "PI_ERROR_$MODULE_$CLASS_NOT_ENOUGH_RECORDED_DATA": {
            "id": 38,
            "status": "released",
            "module": "DataRecorder",
            "class": ["Command"],
            "description": "Not enough recorded data in at least one data recorder trace",
            "internal_usage": "",
            "cmds": ["RECq_DAT"],
            "obsolete_id": 70,
            "obsolete_define": "PI_ERROR_RECORDER_NOT_ENOUGH_RECORDED_DATA",
            "pageId": 225674746,
            "obsolete_error_codes": [25414]
        },
        "PI_ERROR_$MODULE_$CLASS_TRACES_NOT_CONFIGURED": {
            "id": 39,
            "status": "released",
            "module": "DataRecorder",
            "class": ["Command"],
            "description": "Data recorder traces are not configured",
            "internal_usage": "",
            "cmds": ["REC_START"],
            "obsolete_id": 71,
            "obsolete_define": "PI_ERROR_RECORDER_TRACES_NOT_CONFIGURED",
            "pageId": 225674749,
            "obsolete_error_codes": [25415]
        },
        "PI_ERROR_$MODULE_$CLASS_COMMUNICATION_ERROR": {
            "id": 40,
            "status": "released",
            "module": "Com",
            "class": ["Warning"],
            "description": "Controller detected communication error",
            "internal_usage": "",
            "cmds": [],
            "obsolete_id": 100,
            "obsolete_define": "PI_ERROR_COM_COMMUNICATION_ERROR",
            "pageId": 225674725,
            "obsolete_error_codes": [33380]
        },
        "PI_ERROR_$MODULE_$CLASS_FW_INDEX_UNKNOWN": {
            "id": 41,
            "status": "released",
            "module": "Com",
            "class": ["Command"],
            "description": "Firmware index unknown",
            "internal_usage": "",
            "cmds": ["FWC","FWCq","FWE","FWEq","FWRq"],
            "obsolete_id": 2,
            "obsolete_define": "PI_ERROR_COM_FW_INDEX_UNKNOWN",
            "pageId": 225674726,
            "obsolete_error_codes": [33538]
        },
        "PI_ERROR_$MODULE_$CLASS_TIMEOUT": {
            "id": 42,
            "status": "released",
            "module": "Com",
            "class": ["Critical"],
            "description": "Communication timeout occurred",
            "internal_usage": "",
            "cmds": [],
            "obsolete_id": 4,
            "obsolete_define": "PI_ERROR_COM_TIMEOUT",
            "pageId": 225674728,
            "obsolete_error_codes": [33796]
        },
        "PI_ERROR_$MODULE_$CLASS_INVALID_SOCKET": {
            "id": 43,
            "status": "released",
            "module": "Com",
            "class": ["Critical"],
            "description": "Invalid Socket",
            "internal_usage": "",
            "cmds": [],
            "obsolete_id": 11,
            "obsolete_define": "PI_ERROR_COM_INVALID_SOCKET",
            "pageId": 225674727,
            "obsolete_error_codes": [33803]
        },
        "PI_ERROR_$MODULE_$CLASS_WRONG_UNIT_ID_FORMAT": {
            "id": 44,
            "status": "released",
            "module": "System",
            "class": ["Debug"],
            "description": "Wrong unit ID format given",
            "internal_usage": "",
            "cmds": [],
            "obsolete_id": 1,
            "obsolete_define": "PI_ERROR_SYS_WRONG_UNIT_ID_FORMAT",
            "pageId": 225674775,
            "obsolete_error_codes": [36865]
        },
        "PI_ERROR_$MODULE_$CLASS_UNIT_NOT_INITIALIZED": {
            "id": 45,
            "status": "released",
            "module": "System",
            "class": ["Debug"],
            "description": "Unit not initialized",
            "internal_usage": "",
            "cmds": [],
            "obsolete_id": 3,
            "obsolete_define": "PI_ERROR_SYS_UNIT_NOT_INITIALIZED",
            "pageId": 225674770,
            "obsolete_error_codes": [36867]
        },
        "PI_ERROR_$MODULE_$CLASS_MAX_CONNECTION_NUMBER_REACHED": {
            "id": 46,
            "status": "released",
            "module": "System",
            "class": ["Debug"],
            "description": "The max number of storable connections has been reached",
            "internal_usage": "",
            "cmds": [],
            "obsolete_id": 7,
            "obsolete_define": "PI_ERROR_SYS_MAX_CONNECTION_NUMBER_REACHED",
            "pageId": 225674764,
            "obsolete_error_codes": [36871]
        },
        "PI_ERROR_$MODULE_$CLASS_CONNECTION_OUTPUT_WRONG_ARGUMENTS": {
            "id": 47,
            "status": "released",
            "module": "System",
            "class": ["Debug"],
            "description": "Wrong arguments when setting a connection output",
            "internal_usage": "",
            "cmds": [],
            "obsolete_id": 10,
            "obsolete_define": "PI_ERROR_SYS_CONNECTION_OUTPUT_WRONG_ARGUMENTS",
            "pageId": 225674755,
            "obsolete_error_codes": [36874]
        },
        "PI_ERROR_$MODULE_$CLASS_CONNECTION_INPUT_WRONG_ARGUMENTS": {
            "id": 48,
            "status": "released",
            "module": "System",
            "class": ["Debug"],
            "description": "Wrong arguments when setting a connection input",
            "internal_usage": "",
            "cmds": [],
            "obsolete_id": 11,
            "obsolete_define": "PI_ERROR_SYS_CONNECTION_INPUT_WRONG_ARGUMENTS",
            "pageId": 225674752,
            "obsolete_error_codes": [36875]
        },
        "PI_ERROR_$MODULE_$CLASS_WRONG_DEVICE_ID": {
            "id": 49,
            "status": "released",
            "module": "System",
            "class": ["Debug"],
            "description": "Wrong device ID",
            "internal_usage": "",
            "cmds": [],
            "obsolete_id": 13,
            "obsolete_define": "PI_ERROR_SYS_WRONG_DEVICE_ID",
            "pageId": 225674772,
            "obsolete_error_codes": [36877]
        },
        "PI_ERROR_$MODULE_$CLASS_WRONG_FUNCTION_ID": {
            "id": 50,
            "status": "released",
            "module": "System",
            "class": ["Debug"],
            "description": "Wrong function ID",
            "internal_usage": "",
            "cmds": [],
            "obsolete_id": 14,
            "obsolete_define": "PI_ERROR_SYS_WRONG_FUNCTION_ID",
            "pageId": 225674773,
            "obsolete_error_codes": [36878]
        },
        "PI_ERROR_$MODULE_$CLASS_WRONG_PROXY_ID": {
            "id": 51,
            "status": "released",
            "module": "System",
            "class": ["Debug"],
            "description": "Wrong proxy ID",
            "internal_usage": "",
            "cmds": [],
            "obsolete_id": 15,
            "obsolete_define": "PI_ERROR_SYS_WRONG_PROXY_ID",
            "pageId": 225674774,
            "obsolete_error_codes": [36879]
        },
        "PI_ERROR_$MODULE_$CLASS_CONNECTION_OUTPUT_INDEX_OUT_OF_RANGE": {
            "id": 52,
            "status": "released",
            "module": "System",
            "class": ["Debug"],
            "description": "Connection output index is out of range",
            "internal_usage": "",
            "cmds": [],
            "obsolete_id": 40,
            "obsolete_define": "PI_ERROR_SYS_CONNECTION_OUTPUT_INDEX_OUT_OF_RANGE",
            "pageId": 225674754,
            "obsolete_error_codes": [36904]
        },
        "PI_ERROR_$MODULE_$CLASS_INTERFACE_REGISTRATION_FAILED": {
            "id": 53,
            "status": "released",
            "module": "System",
            "class": ["Debug"],
            "description": "Failed to register an interface in the bus gateway",
            "internal_usage": "",
            "cmds": [],
            "obsolete_id": 50,
            "obsolete_define": "PI_ERROR_SYS_INTERFACE_REGISTRATION_FAILED",
            "pageId": 225674763,
            "obsolete_error_codes": [36914]
        },
        "PI_ERROR_$MODULE_$CLASS_DEVICE_REGISTRATION_FAILED": {
            "id": 54,
            "status": "released",
            "module": "System",
            "class": ["Debug"],
            "description": "Failed to register a device in the bus gateway",
            "internal_usage": "",
            "cmds": [],
            "obsolete_id": 51,
            "obsolete_define": "PI_ERROR_SYS_DEVICE_REGISTRATION_FAILED",
            "pageId": 225674757,
            "obsolete_error_codes": [36915]
        },
        "PI_ERROR_$MODULE_$CLASS_PROXY_REGISTRATION_FAILED": {
            "id": 55,
            "status": "released",
            "module": "System",
            "class": ["Debug"],
            "description": "Failed to register a proxy in the bus gateway",
            "internal_usage": "",
            "cmds": [],
            "obsolete_id": 52,
            "obsolete_define": "PI_ERROR_SYS_PROXY_REGISTRATION_FAILED",
            "pageId": 225674767,
            "obsolete_error_codes": [36916]
        },
        "PI_ERROR_$MODULE_$CLASS_INPUT_PORT_ALREADY_CONNECTED": {
            "id": 56,
            "status": "released",
            "module": "System",
            "class": ["Info"],
            "description": "The given input port has already been registered in another connection",
            "internal_usage": "",
            "cmds": [],
            "obsolete_id": 20,
            "obsolete_define": "PI_ERROR_SYS_INPUT_PORT_ALREADY_CONNECTED",
            "pageId": 225674761,
            "obsolete_error_codes": [37140]
        },
        "PI_ERROR_$MODULE_$CLASS_UNIT_ALREADY_REGISTERED": {
            "id": 57,
            "status": "released",
            "module": "System",
            "class": ["Info"],
            "description": "The given unit has already been registered to the broker",
            "internal_usage": "",
            "cmds": [],
            "obsolete_id": 21,
            "obsolete_define": "PI_ERROR_SYS_UNIT_ALREADY_REGISTERED",
            "pageId": 225674768,
            "obsolete_error_codes": [37141]
        },
        "PI_ERROR_$MODULE_$CLASS_CONNECTION_HAS_NO_INPUT": {
            "id": 58,
            "status": "released",
            "module": "System",
            "class": ["Info"],
            "description": "The connection has no input",
            "internal_usage": "",
            "cmds": [],
            "obsolete_id": 60,
            "obsolete_define": "PI_ERROR_SYS_CONNECTION_HAS_NO_INPUT",
            "pageId": 225674750,
            "obsolete_error_codes": [37180]
        },
        "PI_ERROR_$MODULE_$CLASS_CONNECTION_HAS_NO_OUTPUT": {
            "id": 59,
            "status": "released",
            "module": "System",
            "class": ["Info"],
            "description": "The connection has no output",
            "internal_usage": "",
            "cmds": [],
            "obsolete_id": 61,
            "obsolete_define": "PI_ERROR_SYS_CONNECTION_HAS_NO_OUTPUT",
            "pageId": 225674751,
            "obsolete_error_codes": [37181]
        },
        "PI_ERROR_$MODULE_$CLASS_CONNECTION_NOT_FOUND": {
            "id": 60,
            "status": "released",
            "module": "System",
            "class": ["Info"],
            "description": "The sought connection is not registered",
            "internal_usage": "",
            "cmds": [],
            "obsolete_id": 63,
            "obsolete_define": "PI_ERROR_SYS_CONNECTION_NOT_FOUND",
            "pageId": 225674753,
            "obsolete_error_codes": [37183]
        },
        "PI_ERROR_$MODULE_$CLASS_INPUT_PORT_NOT_CONNECTED": {
            "id": 61,
            "status": "released",
            "module": "System",
            "class": ["Info"],
            "description": "The input port is not connected",
            "internal_usage": "",
            "cmds": [],
            "obsolete_id": 64,
            "obsolete_define": "PI_ERROR_SYS_INPUT_PORT_NOT_CONNECTED",
            "pageId": 225674762,
            "obsolete_error_codes": [37184]
        },
        "PI_ERROR_$MODULE_$CLASS_DATA_CORRUPT": {
            "id": 62,
            "status": "released",
            "module": "System",
            "class": ["Warning"],
            "description": "Data corrupted",
            "internal_usage": "",
            "cmds": [],
            "obsolete_id": 70,
            "obsolete_define": "PI_ERROR_SYS_DATA_CORRUPT",
            "pageId": 225674756,
            "obsolete_error_codes": [37446]
        },
        "PI_ERROR_$MODULE_$CLASS_UNIT_TYPE_NOT_SUPPORTED": {
            "id": 63,
            "status": "released",
            "module": "System",
            "class": ["Command", "Critical"],
            "description": "Unit type not supported",
            "internal_usage": "",
            "cmds": ["STVq"],
            "obsolete_id": 1,
            "obsolete_define": "PI_ERROR_SYS_CMD_UNIT_TYPE_NOT_SUPPORTED",
            "pageId": 243503354,
            "obsolete_error_codes": [37633]
        },
        "PI_ERROR_$MODULE_$CLASS_FW_UPDATE_ERROR": {
            "id": 64,
            "status": "released",
            "module": "System",
            "class": ["Command"],
            "description": "Firmware update failed",
            "internal_usage": "",
            "cmds": ["FWEq"],
            "obsolete_id": 50,
            "obsolete_define": "PI_ERROR_SYS_FW_UPDATE_ERROR",
            "pageId": 225674760,
            "obsolete_error_codes": [37682]
        },
        "PI_ERROR_$MODULE_$CLASS_UNIT_NOT_FOUND": {
            "id": 65,
            "status": "released",
            "module": "System",
            "class": ["Command"],
            "description": "Unit could not be found",
            "internal_usage": "Should only occur during internal development, or as a serious error (e.g. one of the module cards dies completely)",
            "cmds": [],
            "obsolete_id": 60,
            "obsolete_define": "PI_ERROR_SYS_UNIT_NOT_FOUND",
            "pageId": 225674769,
            "obsolete_error_codes": [37692]
        },
        "PI_ERROR_$MODULE_$CLASS_CUNIT_NOT_FOUND": {
            "id": 66,
            "status": "released",
            "module": "System",
            "class": ["Command"],
            "description": "Container unit could not be found",
            "internal_usage": "",
            "cmds": ["CPA","REC_TRACE","SPC","SPV","SPVq","STVq","USGq_PAM","USGq_PROP","USGq_SYS"],
            "obsolete_id": 61,
            "obsolete_define": "PI_ERROR_SYS_CUNIT_NOT_FOUND",
            "pageId": 240060897,
            "obsolete_error_codes": [37693]
        },
        "PI_ERROR_$MODULE_$CLASS_FUNIT_NOT_FOUND": {
            "id": 67,
            "status": "released",
            "module": "System",
            "class": ["Command"],
            "description": "Function unit could not be found",
            "internal_usage": "",
            "cmds": ["CPA","REC_TRACE","SPC","SPV","SPVq","USGq_PAM","USGq_PROP","USGq_SYS"],
            "obsolete_id": 62,
            "obsolete_define": "PI_ERROR_SYS_FUNIT_NOT_FOUND",
            "pageId": 240060898,
            "obsolete_error_codes": [37694]
        },
        "PI_ERROR_$MODULE_$CLASS_NOT_ENOUGH_MEMORY": {
            "id": 68,
            "status": "released",
            "module": "System",
            "class": ["Critical"],
            "description": "Not enough memory",
            "internal_usage": "",
            "cmds": [],
            "obsolete_id": 30,
            "obsolete_define": "PI_ERROR_SYS_NOT_ENOUGH_MEMORY",
            "pageId": 225674766,
            "obsolete_error_codes": [37918]
        },
        "PI_ERROR_$MODULE_$CLASS_FLASH_READ_FAILED": {
            "id": 69,
            "status": "released",
            "module": "System",
            "class": ["Critical"],
            "description": "Flash read failed",
            "internal_usage": "",
            "cmds": [],
            "obsolete_id": 50,
            "obsolete_define": "PI_ERROR_SYS_FLASH_READ_FAILED",
            "pageId": 225674759,
            "obsolete_error_codes": [37938]
        },
        "PI_ERROR_$MODULE_$CLASS_NO_DATA_AVAILABLE": {
            "id": 70,
            "status": "released",
            "module": "System",
            "class": ["Critical"],
            "description": "No data available",
            "internal_usage": "",
            "cmds": ["CPA"],
            "obsolete_id": 60,
            "obsolete_define": "PI_ERROR_SYS_NO_DATA_AVAILABLE",
            "pageId": 225674765,
            "obsolete_error_codes": [37948]
        },
        "PI_ERROR_$MODULE_$CLASS_FATAL_ERROR": {
            "id": 71,
            "status": "released",
            "module": "System",
            "class": ["Critical"],
            "description": "Fatal system error occurred",
            "internal_usage": "",
            "cmds": [],
            "obsolete_id": 100,
            "obsolete_define": "PI_ERROR_SYS_FATAL_ERROR",
            "pageId": 225674758,
            "obsolete_error_codes": [37988]
        },
        "PI_ERROR_$MODULE_$CLASS_AXIS_IN_FAULT": {
            "id": 72,
            "status": "released",
            "module": "Motion",
            "class": ["Command"],
            "description": "Motion axis in fault state",
            "internal_usage": "",
            "cmds": ["SAM","MOV","OCV","SMV","FRF","EAX","HLT","SVO"],
            "obsolete_id": "",
            "obsolete_define": "",
            "pageId": 0,
            "obsolete_error_codes": []
        },
        "PI_ERROR_$MODULE_$CLASS_REF_SIGNAL_NOT_FOUND": {
            "id": 73,
            "status": "released",
            "module": "Motion",
            "class": ["Motion"],
            "description": "No reference signal detected",
            "internal_usage": "",
            "cmds": [],
            "pageId": 0
        },
        "PI_ERROR_SYS_$CLASS_TIMEOUT": {
            "id": 74,
            "status": "released",
            "module": "System",
            "class": ["Critical"],
            "description": "Timeout occurred",
            "internal_usage": "",
            "cmds": [],
            "pageId": 0
        },
        "PI_ERROR_$MODULE_$CLASS_ON_IPR": {
            "id": 75,
            "status": "released",
            "module": "Motion",
            "class": ["Command", "Info"],
            "description": "Action not possible due to internal process running",
            "internal_usage": "",
            "cmds": ["SAM","MOV","OCV","SMV","EAX","HLT","SVO"],
            "pageId": 0
        },
        "PI_ERROR_$MODULE_$CLASS_HALT_WAS_COMMANDED": {
            "id": 76,
            "status": "in review",
            "module": "Motion",
            "class": ["Motion"],
            "description": "HALT was triggered by command",
            "internal_usage": "Der Fehler wird vom motioninterface gesetzt wenn ein HLT oder STP kommando bei dieser Achse ausgeführt wurde.",
            "cmds": [],
            "pageId": 0
        },
        "PI_ERROR_$MODULE_$CLASS_SYS_UNIT_TYPE_NOT_SUPPORTED": {
            "id": -1,
            "status": "deprecated",
            "module": "System",
            "class": ["Critical"],
            "description": "Unit type not supported",
            "internal_usage": "",
            "cmds": ["STVq"],
            "obsolete_id": 1,
            "obsolete_define": "PI_ERROR_SYS_UNIT_TYPE_NOT_SUPPORTED",
            "pageId": 225674771,
            "obsolete_error_codes": [37889]
        },
        "PI_ERROR_$MODULE_$CLASS_UNIT_TYPE_WRONG": {
            "id": -1,
            "status": "deprecated",
            "module": "System",
            "class": ["Command"],
            "description": "Wrong unit type",
            "internal_usage": "",
            "cmds": [],
            "obsolete_id": 10,
            "obsolete_define": "PI_ERROR_SYS_UNIT_TYPE_WRONG",
            "pageId": 240060896,
            "obsolete_error_codes": [
                37642
            ]
        }
    }
}